packages/moonstone/Popup/Popup.js

```diff
@@ -383,24 +389,34 @@ class Popup extends React.Component {
 	constructor (props) {
 		super(props);
 		this.paused = new Pause('Popup');
+		const animateOpen = this.props.noAnimation ? OpenState.OPEN : OpenState.OPENING;
 		this.state = {
 			floatLayerOpen: this.props.open,
-			popupOpen: this.props.open,
+			popupOpen: this.props.open ? animateOpen : OpenState.CLOSED,
+			prevOpen: this.props.open,
 			containerId: Spotlight.add(),
 			activator: null
 		};
 		checkScrimNone(this.props);
 	}
 
 	static getDerivedStateFromProps (props, state) {
-		if (props.open && !state.floatLayerOpen) {
-			return {
-				floatLayerOpen: true
-			};
-		} else if (!props.open && state.popupOpen) {
-			return {
-				popupOpen: false
-			};
+		if (props.open !== state.prevOpen) {
+			if (props.open) {
+				return {
+					popupOpen: props.noAnimation ? OpenState.OPEN : OpenState.CLOSED,
```

Comment:
I noticed one situation in which this condition doesn't quite work. This isn't actually a regression from your last update but instead my previous change that introduced `OpenState`. In sampler, if you check the `open` knob, let the animation complete, and then quickly toggle `open` off and then on, the dialog will not be displayed.

From my analysis, it appears that since the floating layer isn't closed (because the popup didn't finish closing) we should check its state here as well and, if `true`, set `OpenState.OPEN`.

```
popupOpen: props.noAnimation || state.floatLayerOpen ? OpenState.OPEN : OpenState.CLOSED,
```

@viodragon2 - can you check this out and see if you come to the same conclusion?

Reply:
Yup. That makes sense. 
On a side note, it would be nice if `Transition` supports mid transition events (e.g. `entering`) so that we can handle these cases.

Reply:
Sure. That's one of the constraints we have with CSS transitions. We only get an "end" event and only if the triggering CSS prop has changed since the last paint/layout (i'm not clear which off hand, tbh).

Reply:
This is something I've been wanting to add since Enyo, but we didn't have a use-case, so I couldn't justify the feature when originally writing Enact's Transition. If we're still interested, let's create a ticket and start this process.